### DISCUSSION TITLE
I have the following view and facing an issue with it any guidance or help is much appreciated

### DISCUSSION THREAD

**Discussion by alialjowaher** [viewflow/viewflow#466] [General]
``` class SubmitRequestView(TaskSuccessUrlMixin, CreateView):
    model = ApprovalProcess
    fields = ['name', 'description']
    template_name = 'core/submit_request.html'

    def form_valid(self, form):
        object = form.save()
        object.save()

        # Keep it as workflow process artifact
        self.request.activation.process.artifact = object

        self.request.activation.execute()
        return redirect(self.get_success_url())```



the error I'm getting is " 'WSGIRequest' object has no attribute 'activation' "

**Comment by darahsten** [comment]
It appears that you have a class based view but have excluded all the code that can help others help you resolve your issue. The request object is likely not to have attribute "activation" if the underlying class based view does not implement the Flow mechanisms. 
I would say check the parent classes and ensure you are inheriting from the correct ones. But that is a wild guess

**Reply by alialjowaher** [reply]
Sorry about that I'm using Django's built in CreateView

**Comment by darahsten** [comment]
That was kind of obvious. You are not making it particularly easy to help you. But in any case, follow this class [https://github.com/viewflow/cookbook/blob/main/workflow101/bloodtest/views.py#L80](url)

**Comment by kmmbvnr** [comment]
Hm, maybe you've added your view directly into URL patterns config instead of flow class urls?

Viewflow wraps views and initialises request.acticlvation in a wrapper code, before actual view call

You own view itself doesn't need any additional code, the request.activation initialized in a wrapper

https://github.com/viewflow/viewflow/blob/main/viewflow/workflow/flow/nodes.py#L43

So when you've added flow urls into URL patterns, when you access /process_pk/task_pk/ the wrapper is called and request.activation is initialized
